File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    count=0
    people.sort()
    start=0
    end=len(people)-1

    while start<=end:
        if people[start]+people[end]<=limit:
            start+=1
            end-=1
        else:
            end-=1
        
        count+=1
    
    return count

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution{
    public int solution(int[] people, int limit){
        Arrays.sort(people);
        
        int left=0;
        int right=people.length-1;
        int boats=0;
        
        while(left<=right){
            //가장 가벼운 사람 + 가장 무거운 사람이 같이 탈 수 있으면?
            if(people[left]+people[right]<=limit){
                left++; //가벼운 사람도 태움
            }
            //무거운 사람은 항상 이번 보트에 탑승
            right--;
            boats++;
        }
        return boats;
    }
}